include/rmm/mr/device/device_memory_resource.hpp

```diff
@@ -238,4 +392,5 @@ class device_memory_resource {
   [[nodiscard]] virtual std::pair<std::size_t, std::size_t> do_get_mem_info(
     cuda_stream_view stream) const = 0;
 };
+static_assert(cuda::mr::resource_with<device_memory_resource, cuda::mr::device_accessible>, "");
```

Comment:
Remind me what this is checking? Perhaps a non-empty string message would help.

Reply:
This will look unfamiliar if one hasn't done much `concept`s programming yet, but this is a common way to check that a type satisfies a concept.

`resource_with` is a `concept` that says "This type is a `resource` _with_ the following properties" (where `resource` is another concept that defines the interface).

Reply:
They joys of working in a C++17 codebase :)

I have removed all empty strings in static_asserts